test/tcp.rs

```diff
@@ -247,3 +247,15 @@ fn listen_then_close() {
     let mut h = H;
     e.run_once(&mut h).unwrap();
 }
+
+#[test]
+fn test_tcp_sockets_are_send() {
+    fn foo<T: Send>(t: T) {
```

Comment:
you could write this test without having to actually create TcpListener's and TcpStreams by just doing:

```rust
fn assert_send<T: Send>() {}
assert_send::<TcpListener>();
assert_send::<TcpStream>();
```